### PR TITLE
LazyTree: moving out Tree.Element from LazyTreeState 

### PR DESCRIPTION
**Pull request by fscarponi**

### CODE REVIEW

File: foundation/src/main/kotlin/org/jetbrains/jewel/foundation/tree/BasicLazyTree.kt [hunk 1]
```diff
@@ -216,3 +223,62 @@ value class TreeElementState(val state: ULong) {
...
+    treeState: TreeState,
+    tree: Tree<*>,
+) {
+    Log.d("-----treeRefreshed----")
+    flattenedTree.value = tree.roots.flatMap { flattenTree(it, treeState.openNodes, treeState.allNodes) }
+    flattenedTree.value.forEach {
+        Log.d(it.idPath().toString())
+    }
+//    delegate.updateKeysIndexes()
+    Log.d("----treeRefreshedPrinted----")
```
Comment (rock3r) [inline foundation/src/main/kotlin/org/jetbrains/jewel/foundation/tree/BasicLazyTree.kt:238]: Can we remove the debug statements and commented code?

File: foundation/src/main/kotlin/org/jetbrains/jewel/foundation/tree/BasicLazyTree.kt [hunk 2]
```diff
@@ -216,3 +223,62 @@ value class TreeElementState(val state: ULong) {
         )
     }
 }
+
+private suspend fun refreshFlattenTree(
```
Comment (rock3r) [inline foundation/src/main/kotlin/org/jetbrains/jewel/foundation/tree/BasicLazyTree.kt:227]: `refreshFlattenedTree()`? It's also very dodgy to pass in a State, I'd pass in the value and output the new one instead, reassigning the new value where the state is held instead.

File: foundation/src/main/kotlin/org/jetbrains/jewel/foundation/tree/TreeState.kt [hunk 3]
```diff
@@ -30,114 +26,94 @@ class TreeState(
...
-                }
-            }
-
-            is Tree.Element.Leaf<*> -> {
-                orderedChildren.add(element)
-            }
-        }
-        return orderedChildren
-    }
+//
```
Comment (rock3r) [inline foundation/src/main/kotlin/org/jetbrains/jewel/foundation/tree/TreeState.kt:35]: Can we delete commented out code?

File: foundation/src/main/kotlin/org/jetbrains/jewel/foundation/tree/TreeState.kt [hunk 4]
```diff
@@ -161,7 +138,13 @@ class TreeState(
         delegate.deselectAll()
     }
 
-    fun isNodeOpen(element: Tree.Element.Node<*>) = element.isOpen
-
-    fun toggleNode(element: Tree.Element.Node<*>) = if (element.isOpen) closeNode(element) else openNode(element)
+    fun toggleNode(nodeId: Any) {
+        Log.d("toggleNode $nodeId")
```
Comment (rock3r) [inline foundation/src/main/kotlin/org/jetbrains/jewel/foundation/tree/TreeState.kt:142]: We should remove these logging statements

**Review by rock3r — COMMENTED** [review]
Just some comments, then good to go

File: foundation/src/main/kotlin/org/jetbrains/jewel/foundation/tree/TreeState.kt (hunk 4 above)
Comment (fscarponi) [reply foundation/src/main/kotlin/org/jetbrains/jewel/foundation/tree/TreeState.kt:142]: the log is set to OFF by default, i will remove it later, when the selectableLazyColumn's focus issue is fixed

**Review by rock3r — APPROVED** [review]
ayeaye

**Pull request merged**